src/datachain/data_storage/warehouse.py

```diff
@@ -218,35 +218,26 @@ def dataset_select_paginated(
         results = None
         offset = 0
         num_yielded = 0
-        try:
-            while True:
-                if limit is not None:
-                    limit -= num_yielded
-                    if limit == 0:
-                        break
-                    if limit < page_size:
-                        paginated_query = paginated_query.limit(None).limit(limit)
-
-                results = self.dataset_rows_select(paginated_query.offset(offset))
-
-                processed = False
-                for row in results:
-                    processed = True
-                    yield row
-                    num_yielded += 1
-
-                if not processed:
-                    break  # no more results
-                offset += page_size
-        finally:
-            # https://www2.sqlite.org/cvstrac/wiki?p=DatabaseIsLocked (SELECT not
-            # finalized or reset) to prevent database table is locked error when an
-            # exception is raised in the middle of processing the results (e.g.
-            # https://github.com/iterative/dvcx/issues/924). Connections close
-            # apparently is not enough in some cases, at least on sqlite
-            # https://www.sqlite.org/c3ref/close.html
-            if results and hasattr(results, "close"):
-                results.close()
+
+        while True:
+            if limit is not None:
+                limit -= num_yielded
+                if limit == 0:
+                    break
+                if limit < page_size:
+                    paginated_query = paginated_query.limit(None).limit(limit)
+
+            results = self.dataset_rows_select(paginated_query.offset(offset))
```

Comment:
https://github.com/iterative/datachain/pull/339 changed the code to 

```python
results =  yield from convert_rows_custom_column_types(
                query.selected_columns, rows, self.db.dialect
            )
```

which I don't think will ever have a `close` attribute.

Reply:
Since the `results` is an iterator, I think we still need to close this.

Reply:
I moved it underneath into `dataset_rows_select` but it doesn't look like it is required there either. The original intention from https://github.com/iterative/dvcx/pull/1369 was to get `test_row_generator_with_new_columns_wrong_type` passing. We can now run the entire suite without having the `finally` statement in either method.